azurerm/internal/services/logic/logic_app_workflow_resource_test.go

```diff
@@ -283,3 +298,66 @@ resource "azurerm_logic_app_workflow" "test" {
 }
 `, IntegrationServiceEnvironmentResource{}.basic(data), data.RandomInteger)
 }
+
+func (LogicAppWorkflowResource) parameters(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-logic-%d"
+  location = "%s"
+}
+
+resource "azurerm_logic_app_workflow" "test" {
+  name                = "acctestlaw-%d"
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+  workflow_parameters = {
+    b = jsonencode({
+      type = "Bool"
+    })
+    str = jsonencode({
+      type = "String"
+    })
+    int = jsonencode({
+      type = "Int"
+    })
+    float = jsonencode({
+      type = "Float"
+    })
+    obj = jsonencode({
+      type = "Object"
+    })
+    secstr = jsonencode({
+      type = "SecureString"
+    })
+    secobj = jsonencode({
+      type = "SecureObject"
+    })
+  }
+
+  parameters = {
+    b     = "true"
+    str   = "value"
+    int   = "123"
+    float = "1.23"
+    obj = jsonencode({
+      foo = "bar"
+    })
+    secstr = "value"
+    secobj = jsonencode({
+      foo = "bar"
+    })
+  }
+
+  lifecycle {
+    ignore_changes = [
+      parameters["secstr"],
+      parameters["secobj"],
+    ]
+  }
```

Comment:
how come we are ignoring changes heres?

Reply:
These are secure types, means they will not return from API on read.

Reply:
so in that case we should be pulling the existing values from the config through into the read function so the user does not have to do this

Reply:
You're right. I've updated the code to read the security params from state, and ignore them in the import verification accordingly. Also I refined the test for the "Array", with some modification on the code.

## Test Result

```
💤 TF_ACC=1 go test -timeout=3h -v ./azurerm/internal/services/logic -run='TestAccLogicAppWorkflow_parameters'
=== RUN   TestAccLogicAppWorkflow_parameters                                                             
=== PAUSE TestAccLogicAppWorkflow_parameters                                                             
=== CONT  TestAccLogicAppWorkflow_parameters                                                             
--- PASS: TestAccLogicAppWorkflow_parameters (92.06s)                  
PASS                                                                                                     
ok      github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/logic       92.080s
```